### ISSUE TITLE
Could not resolve path to config /d:/.../psalm.xml

### ISSUE THREAD

**Issue by glen-84** [psalm/psalm-vscode-plugin#84]
I don't think that the colon is supposed to be there.

Something is converting a Windows path to a Linux path incorrectly, it seems.

This extension worked previously (pre-2.0).

**Comment by tm1000** [comment]
The colon isn't the issue it's the prefixed slash

**Comment by glen-84** [comment]
I only have:

```json
"psalm.phpExecutablePath": "C:\\PHP-7.4.x\\php.exe"
```

Config file is [here](https://github.com/glen-84/phpcs-standard/blob/a015e6070cc96b10fbaca635b46dedd66eaad301/psalm.xml).

**Comment by tm1000** [comment]
Try declaring the path directly through configuration:

```
"psalm.configPaths": ["app/psalm.xml"],
```

**Comment by glen-84** [comment]
It doesn't work even if I use an absolute path like:

```json
    "psalm.configPaths": [
        "D:/Programming/Standards/phpcs-standard/psalm.xml"
    ]
```

> [ERROR - 20:08:16] No Config file found in: D:/Programming/Standards/phpcs-standard/psalm.xml

If I ctrl+click on that path, the file opens in VSC.

**Comment by tm1000** [comment]
The code that is causing this is:

```
const psalmXML = await vscode.workspace.findFiles(
        `{${configPaths.join(',')}}`
        // `**/vendor/**/{${configPaths.join(',')}}`
    );
    if (!psalmXML.length) {
        // no psalm.xml found
        loggingService.logError(
            `No Config file found in: ${configPaths.join(',')}`
        );
        return;
    }
```

`vscode.workspace.findFiles` is an internal function to vscode (I do not control it). For you it's returning 0 results. Will have to investigate this further

**Comment by Ambrosia** [comment]
A less than ideal workaround is to insert the following code in `vendor/vimeo/psalm/src/Psalm/Internal/CliUtils.php:581`:

```php
if(isset($options['c']) && is_string($options['c']) && preg_match('/\/\w:/', $options['c']) === 1) {
    $options['c'] = substr($options['c'], 1);
}
```

This will remove the leading slash if the first path segment is a Windows drive letter. You'll have to add this each time you update psalm.

**Comment by tm1000** [comment]
@Ambrosia is this a psalm issue though? Seems to be with the plug-in for vscode.

**Comment by Ambrosia** [comment]
> @Ambrosia is this a psalm issue though? Seems to be with the plug-in for vscode.

Nah, it's still vscode's fault, not psalm itself. I just wrote this as a quick workaround until it it gets fixed for real. I found it easier to just modify psalm than to figure out how to do it in this extension itself.

**Comment by tm1000** [comment]
Fixed in 2.2.3
